src/doc/en/installation/meson.rst

```diff
@@ -65,11 +65,23 @@ or run the tests with ``./sage -t``.
     This means that any Sage-the-distribution commands such as ``sage -i`` 
     will not work.
 
+.. NOTE::
+
+    By default, ``ninja`` will automatically determine the number of jobs to
+    run in parallel based on the number of CPU available. This can be adjusted
+    by passing ``--config-settings=compile-args=-jN`` to ``pip install``,
+    which will pass ``-jN`` to ``ninja``.
+
+    ``--verbose`` can be passed to ``pip install``, then the meson commands
+    internally used by pip will be printed out.
+
 Background information
 ======================
 
 Under the hood, pip invokes meson to configure and build the project.
 We can also use meson directly as follows.
+``meson compile`` is much faster than ``pip install``,
```

Comment:
Can you explain why someone would need to run multiple times `pip install`? This normally should be a one-time thing.

Maybe add a disclaimer saying that using meson directly can be advantageous in certain cases, e.g. if you want to work on the meson build files.

(Background: The standard setting should be `pip install -e .` - and that's all for almost everyone. If there are issues with this installation method, then they should be fixed.)

Reply:
As in if you modify the `.pyx` file you need some way to recompile… but I realize the editable loader automatically recompiles if needed. Okay that can be mentioned in the documentation.

(for the original question: I use meson as a replacement of monolithic build system for development nowadays. It's faster because it eliminates the need of recompiling dependent packages on large updates.)